### PR TITLE
Enable Automatic Mixed Precision and update training settings

### PR DESCRIPTION
**Pull request by Saad-Musema**
This PR introduces **Automatic Mixed Precision (AMP)** for training the PC Transformer, reducing GPU memory usage and slightly speeding up training.

### Commit Reference
Related commit: [cbed1fe](https://github.com/iCog-Labs-Dev/PC-Transformers/commit/cbed1fe1daf8f7677704d48556a84d6fff4bba4c)

### Configurations used to compare updates with and without AMP:
- **Hyperparameters**:
  - `block_size = 256`
  - `warmup_steps = 2`
  - `n_embed = 64`
  - `T = 2`
  - `num_heads = 2`
  - `n_blocks = 2`
  - `num_epochs = 2`
  - `batch_size = 8`
  
- **Dataloader**: set to load only a 0.001 sample ratio for faster testing

### Results Comparison:

| Metric | With AMP | Without AMP | Difference |
|--------|----------|------------|------------|
| Peak memory (MB) | 2427 | 2960 | -18% less memory |
| Training time (s) | 2069.6 | 2101.4 | ~1.5% faster |

### CODE REVIEW

**Comment by Saad-Musema** [comment]
I have made sure to test the code with the hyperparameter configuration listed above, and I removed additional commits that I used to test the memory allocation
